### DISCUSSION TITLE
course/starbook/word-list

### DISCUSSION THREAD

**Discussion by giscus** [gogf/gf-site#65] [General]
# course/starbook/word-list

使用GET方式查询单词的分页列表，包含模糊查询功能。定义结构体保存单词字段，包括id、单词、定义及熟练度。在GoFrame框架中编写Logic，用于处理数据的查询和分页操作。通过Controller调用Logic，实现数据的获取和返回，支持综合的接口测试。

https://goframe.org/course/starbook/word-list

**Comment by cbtan** [comment]
MSSQL 数据库分页失败：

{
	"code": 52,
	"message": "SELECT * FROM words WHERE uid=8 ORDER BY created_at desc,id desc OFFSET 0 ROWS FETCH NEXT 15 ROWS ONLY: mssql: 在 FETCH 语句中选项 NEXT 的用法无效。",
	"data": null
}

**Reply by oldme-git** [reply]
`sql server 2008`不支持 `NEXT` 语句，请尝试升级到2012

**Reply by cbtan** [reply]
确实，我换了2019后可以了！感谢！！！

**Comment by Grace-Cao** [comment]
ProficiencyLevel: model.ProficiencyLevel(v.ProficiencyLevel),  

该行提示：  ProficiencyLevel: v.ProficiencyLevel,  
Is OK

**Reply by oldme-git** [reply]
取决你 List ProficiencyLevel 的字段类型

**Reply by Grace-Cao** [reply]
提示：文件api/words/v1/words_struct.go内，
type List struct {  
    Id               uint        `json:"id"`  
    Word             string      `json:"word"`  
    Definition       string      `json:"definition"`  
    ProficiencyLevel uint        `json:"proficiencyLevel"`
}
ProficiencyLevel 类型应为：model.ProficiencyLevel类型

**Reply by RichZDS** [reply]
虽然有点晚了 但是这个model.ProficiencyLevel 上面写的代码有一点错误 应该是ProficiencyLevel  因为model包里面没有ProficiencyLevel字段 只有这个words里面有

**Comment by chaegumi** [comment]
很多示例代码需要更新一下，比如logic中定义type，gf gen service的时候会报错的，需要定义到api中去
